content/blog/2022/2022-09-02-junit5.md

```diff
@@ -0,0 +1,601 @@
+	    assertEquals(expectedString, "woof1");  
+	    System.out.println("WOOF => 1");  
+	}  
+	  
+	@Test  
+	public void testBark2() {  
+	    String expectedString = "woof2";  
+	    assertEquals(expectedString, "woof2");  
+	    System.out.println("WOOF => 2");  
+	}
+}
+```
+We should see the test not running and the disabled message being displayed.
+```
+Dog 1 please don't woof
+WOOF => 2
+```
+
+## Assertions
+JUnit offers basic assertions through the Jupiter module. Let's explore the main ones. For the full list please check [JUnit's doc](https://junit.org/junit5/docs/5.0.1/api/org/junit/jupiter/api/Assertions.html).
+
+### assertEquals
+This method compares two given parameters for equality . It can take any of the primitive types (int, float, etc...) plus accepting Objects for which it calls the `equals` method to check the equality.
+We have seen many examples of it in our previous code snippets.
+
+### assertNotEquals
+It checks for non-equality for two given parameters (primitive or objects).
+```java
+public class DogTest { 
+	@Test  
+	public void testNotBark() {  
+	    String unexpectedString = "";  
+	    assertNotEquals(unexpectedString, "woof");  
+	    System.out.println("Didn't woof!!");  
+	}
+}
+```
+This should pass and printout
+```
+Didn't woof!!
+```
+
+### assertNull
+It simply does a null check on the given parameter.
+```java
+public class DogTest { 
+	@Test  
+	public void nullCheck() {  
+	    Object dog = null;  
+		assertNull(dog);  
+	    System.out.println("Null dog :(");  
+	}
+}
+```
+This will print
+```
+Null dog :(
+```
+### assertNotNull
+Unlike, `assertNull`, this fails if the given object isn't null;
+```java
+public class DogTest { 
+	@Test  
+	public void nonNullCheck() {  
+	    String dog = "Max";  
+	    assertNotNull(dog);  
+	    System.out.println("Hey I am " + dog);  
+	}
+}
+```
+This will print
+```
+Hey I am Max
+```
+
+### assertTrue
+It will fail if the given boolean parameter is false.
+```java
+public class DogTest { 
+	@Test  
+	public void trueCheck() {  
+	    int dogAge = 2;  
+	    assertTrue(dogAge < 5);  
+	    System.out.println("I am young :)");  
+	}
+}
+```
+This will printout
+```
+I am young :)
+```
+### assertFalse
+It will fail if the given boolean parameters is true.
+```java
+public class DogTest { 
+	@Test  
+	public void falseCheck() {  
+	    int dogAge = 7;  
+	    assertFalse(dogAge < 5);  
+	    System.out.println("I am old :(");  
+	}
+}
+```
+This should show us
+```
+I am old :(
+```
+
+## Assertions with Hamcrest
+The first question that comes to mind is, why do we need an extra dependency to do the assertions when JUnit offers us a basic set of assertions?
+The answer is that the creators of JUnit5 intended it to be a powerful platform for running and managing tests and they didn't want to over-complicate it with a full capability of assertions. Instead, they designed the platform to be fully integrate-able with third party libraries such as Hamcrest.
```

Comment:
Maybe pull up the sentence about the fluent API to here and add it as an argument for using Hamcrest over native JUnit assertions. Maybe add something like this: "As you will see in the code examples below, Hamcrest assertions make the tests much more readable because they read almost like natural sentences"